### PR TITLE
Test beta version of corrosion

### PR DESCRIPTION
**Pull request by ogoffart**

### CODE REVIEW

**Comment by tronical** [comment]
This will also need a change to remove the no_soname property, as corrosion sets it now. (CI will pass anyway)

**Comment by tronical** [comment]
I’m very much in favor of testing this. I think corrosion will release before our next release, right?

File: api/cpp/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ include(FetchContent)
 FetchContent_Declare(
     Corrosion
     GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
-    GIT_TAG v0.3.4
+    GIT_TAG v0.4-beta1
```
Comment (ogoffart) [inline api/cpp/CMakeLists.txt:14]:
```suggestion
    GIT_TAG v0.4-beta2
```

**Comment by tronical** [comment]
The Corrosion update will also fix https://github.com/slint-ui/slint-cpp-template/issues/12 because Corrosion sets `IMPORTED_NO_SONAME` now, which means that CMake will link binaries with `-L/path/to/dir -Wl,rpath,/path/to/dir -llibslint_cpp` instead of `path/to/dir/libslint_cpp.so`. The latter means a path with slashes is encoded in the binary, which means all rpath settings are ignored (as per ld.so man-page).

**Comment by ogoffart** [comment]
Merging the beta version in master because this will increase testing.

**Pull request merged**